src/Ordering.API/Program.cs

```diff
@@ -1,19 +1,19 @@
 ﻿var builder = WebApplication.CreateBuilder(args);
 
 builder.AddServiceDefaults();
-builder.AddDefaultOpenApi();
 builder.AddApplicationServices();
-
 builder.Services.AddProblemDetails();
 
-var app = builder.Build();
+var apiVersioning = builder.Services.AddApiVersioning();
 
-app.UseDefaultOpenApi();
+builder.AddDefaultOpenApi(apiVersioning);
+
+var app = builder.Build();
 
 app.MapDefaultEndpoints();
 
-app.MapGroup("/api/v1/orders")
-   .MapOrdersApi()
-   .RequireAuthorization();
+app.NewVersionedApi("Orders")
+   .MapOrdersApiV1();
 
+app.UseDefaultOpenApi();
```

Comment:
Nit: Can we keep the `UseDefaultOpenApi` call before the `MapX` calls for consistency?

Reply:
Unfortunately - no. This _limitation_ was one of the most infuriating part of _Minimal APIs_ (for me) when they were introduced. I understand why it works the way it does, but when all of the pieces are put together, things only work if configured in a very specific order. The issues are a perfect storm of all the libraries involved.

### Issue 1

ASP.NET Core backs `WebApplication` with an `EndpointDataSource` that it puts into the DI container. This is great for deferred evaluation since the backig services are designed to handle changes to the underlying endpoint collections.

https://github.com/dotnet/aspnetcore/blob/dc1acba9cd1374a8a8560bee655682e1a72de3eb/src/Http/Routing/src/DependencyInjection/RoutingServiceCollectionExtensions.cs#L75

### Issue 2

The API Explorer extensions provided API Versioning define the `IApiVersionDescriptionProvider` service, which is responsible for collating all of the API versions in the application. If all of the `Endpoint`s are deprecated, it will also handle reporting that a given API version is deprecated. This service is used to enumerate all API versions in an application. A common usage scenario is dynamically creating an OpenAPI document for each API version.

The `IApiVersionDescriptionProvider` service is backed by a collection of `IApiVersionMetadataCollationProvider` services that know how to collate API versions across _Minimal APIs_ and good 'ol controllers. These are all in the DI container and backed by the registered `EndpointDataSource` and `IActionDescriptorCollectionProvider`, respectively. At this point, there are no `Endpoint`s registered with `EndpointDataSource`; therefore, the _Minimal API_ collator doesn't discover anything. These services are aware that the underlying sources can change and will adapt, but the changes have to happen first.

https://github.com/dotnet/aspnet-api-versioning/blob/22a775b8eae36147587faaa062d7555260c94736/src/AspNetCore/WebApi/src/Asp.Versioning.Mvc.ApiExplorer/IEndpointRouteBuilderExtensions.cs#L27

### Issue 3

When you finally call `UseSwaggerUI()`, Swashbuckle uses eager evaluation. It's building the middleware for the pipeline and the DI container has already been built, so it isn't that unreasonable.

https://github.com/domaindrivendev/Swashbuckle.AspNetCore/blob/8f363f7359cb1cb8fa5de5195ec6d97aefaa16b3/src/Swashbuckle.AspNetCore.SwaggerUI/SwaggerUIBuilderExtensions.cs#L34

As a result, when the setup for `SwaggerUIOptions` is evaluated in `UseSwaggerUI`:

1. there are no _Minimal APIs_ registered
2. API Versioning doesn't see or collate any API versions from _Minimal APIs_ - yet
3. one or more OpenAPI documents are not generated (which is wrong and unexpected)

-----

You might be wondering why this hasn't been a more prevalent issue. If you're **not** using API Versioning, then it's not a problem because there's only a single OpenAPI document or you've hard-coded the mapping in the configuration setup. Once you want to use the API Versioning API Explorer extensions to drive the configuration automatically from the application, things start to fall down if they are not done in the correct order.

I've had more than one customer file an issue about this behavior. I always direct them to the API Versioning example projects. The differences in the example setups versus the vanilla out-of-the-box templates are only meant to vary by what is absolutely necessary to get things running, not my personal dogma about how I think the template should be structured.

From my experience, combining all of these concepts together requires `UseDefaultOpenApi ` be the last thing or nearly the last thing in the configuration. It definitely needs to be after `MapXXX` calls because that is the only way for API Versioning to discover them (automatically).

We wont' solve the problem here, but I fully support making this experience better. The order should not really matter. This behavior is something you probably want to account for in the forthcoming OpenAPI extensions that are planned for .NET 9.